static/app/components/modals/sudoModal.tsx

```diff
@@ -44,76 +54,94 @@ type Props = WithRouterProps &
     retryRequest?: () => Promise<any>;
   };
 
-type State = {
-  authenticators: Array<Authenticator>;
-  busy: boolean;
-  error: boolean;
-  errorType: string;
-  showAccessForms: boolean;
-  superuserAccessCategory: string;
-  superuserReason: string;
-};
-
-class SudoModal extends Component<Props, State> {
-  static defaultProps: DefaultProps = {
-    closeButton: true,
-  };
-
-  state: State = {
-    authenticators: [],
-    busy: false,
+function SudoModal({
+  closeModal,
+  isSuperuser,
+  location,
+  needsReload,
+  router,
+  retryRequest,
+  api,
+  Header,
+  Body,
+  closeButton,
+}: Props) {
+  const [state, setState] = useState<State>({
+    authenticators: [] as Authenticator[],
     error: false,
     errorType: '',
     showAccessForms: true,
     superuserAccessCategory: '',
     superuserReason: '',
-  };
+  });
+
+  const {
+    authenticators,
+    error,
+    errorType,
+    showAccessForms,
+    superuserAccessCategory,
+    superuserReason,
+  } = state;
+
+  const isLoading = useContext(OrganizationLoaderContext)?.isLoading;
```

Comment:
```suggestion
  const  { isLoading } = useContext(OrganizationLoaderContext);
```
nit: Recommend destructing the value (if there's a type error you can do ` || {}` to get around that)